Core/include/Acts/Vertexing/HelicalTrackLinearizer.ipp

```diff
@@ -14,179 +14,197 @@ Acts::Result<Acts::LinearizedTrack> Acts::
-    positionAtPCA[ePos1] = pos[ePos1];
-    positionAtPCA[ePos2] = pos[ePos2];
     parCovarianceAtPCA = params.covariance().value();
   }
 
-  // phiV and functions
-  double phiV = paramsAtPCA(BoundIndices::eBoundPhi);
-  double sinPhiV = std::sin(phiV);
-  double cosPhiV = std::cos(phiV);
+  // Extracting Perigee parameters and compute functions of them for later
+  // usage
+  double phi = paramsAtPCA(BoundIndices::eBoundPhi);
+  double sinPhi = std::sin(phi);
+  double cosPhi = std::cos(phi);
 
-  // theta and functions
-  double th = paramsAtPCA(BoundIndices::eBoundTheta);
-  const double sinTh = std::sin(th);
-  const double tanTh = std::tan(th);
+  double theta = paramsAtPCA(BoundIndices::eBoundTheta);
+  const double sinTheta = std::sin(theta);
+  const double tanTheta = std::tan(theta);
 
   // q over p
   double qOvP = paramsAtPCA(BoundIndices::eBoundQOverP);
+
+  // Charge of the particle, determines the sign of the helix radius rho
   double sgnH = (qOvP < 0.) ? -1 : 1;
 
-  Vector3 momentumAtPCA(phiV, th, qOvP);
+  // Momentu at the PCA
+  Vector3 momentumAtPCA(phi, theta, qOvP);
 
-  // get B-field z-component at current position
-  auto field = m_cfg.bField->getField(VectorHelpers::position(positionAtPCA),
-                                      state.fieldCache);
+  // Define Jacobians, which will be filled later
+  ActsMatrix<eBoundSize, 4> positionJacobian;
+  positionJacobian.setZero();
+  ActsMatrix<eBoundSize, 3> momentumJacobian;
+  momentumJacobian.setZero();
+  // get the z-component of the B-field at the PCA
+  auto field =
+      m_cfg.bField->getField(VectorHelpers::position(pca), state.fieldCache);
   if (!field.ok()) {
     return field.error();
   }
   double Bz = (*field)[eZ];
-  double rho = 0;
-  // Curvature is infinite w/o b field
-  if (Bz == 0. || std::abs(qOvP) < m_cfg.minQoP) {
-    rho = sgnH * m_cfg.maxRho;
-  } else {
-    rho = sinTh * (1. / qOvP) / Bz;
-  }
-
-  // Eq. 5.34 in Ref(1) (see .hpp)
-  double X = positionAtPCA(0) - linPointPos.x() + rho * sinPhiV;
-  double Y = positionAtPCA(1) - linPointPos.y() - rho * cosPhiV;
-  const double S2 = (X * X + Y * Y);
-  const double S = std::sqrt(S2);
-
-  /// F(V, p_i) at PCA in Billoir paper
-  /// (see FullBilloirVertexFitter.hpp for paper reference,
-  /// Page 140, Eq. (2) )
-  BoundVector predParamsAtPCA;
-
-  int sgnX = (X < 0.) ? -1 : 1;
-  int sgnY = (Y < 0.) ? -1 : 1;
 
-  double phiAtPCA = 0;
-  if (std::abs(X) > std::abs(Y)) {
-    phiAtPCA = sgnH * sgnX * std::acos(-sgnH * Y / S);
-  } else {
-    phiAtPCA = std::asin(sgnH * X / S);
-    if ((sgnH * sgnY) > 0) {
-      phiAtPCA = sgnH * sgnX * M_PI - phiAtPCA;
-    }
-  }
-
-  // Eq. 5.33 in Ref(1) (see .hpp)
-  predParamsAtPCA[0] = rho - sgnH * S;
-  predParamsAtPCA[1] =
-      positionAtPCA[eZ] - linPointPos.z() + rho * (phiV - phiAtPCA) / tanTh;
-  predParamsAtPCA[2] = phiAtPCA;
-  predParamsAtPCA[3] = th;
-  predParamsAtPCA[4] = qOvP;
-  predParamsAtPCA[5] = 0.;
```

Comment:
That we set all time values to zero you mean?

Reply:
as far as I understand the derivatives are basically the same as before but this one is filled with the track time now. so I could imagine that the fit later used this time and since it was 0 all the time it would not match up with the track or vertex time

Reply:
If I replace line 123 with 
```c++
predParamsAtPCA[5] = paramsAtPCA[5].;
``` 
which should reproduce the same result as the new code, AMVF+time still fails...

Reply:
So I guess this is not the only reason why the AMVF+time works now. Maybe numerical instabilities play a role? They affect the old code for large helix radii rho - do you know what a typical value of rho is?

Reply:
hm that is odd. looking at the code it can only be down to the derivatives now?

I think it would be good to know why it works now

in principle a small kick in the derivatives could have moved us over the bar of accepting the vertices but that sound kind of unlikely

Reply:
I think we are using tracks with pT 1-10 GeV in our tests so rho would be on the 1-10 m scale

Reply:
Ok then I think numerical instabilities should not me much of an issue... I will dig a bit deeper to see where the differences are exactly!

Reply:
The AMVF+time can be made working with the old code by changing line 118-119 to `predParamsAtPCA[1] = positionAtPCA[eZ] + rho * (phiV - phiAtPCA) / tanTh;` and line 120 to `predParamsAtPCA[5] = paramsAtPCA[5];`.  In the picture below one sees that predParamsAtPCA uses the z-coordinate wrt the Perigee, while paramsAtPCA uses the global z-coordinate (positionAtPCA[eZ]). It's strange that the propation to the PCA gives a global z-coordinate - might there be a bug with the line surface?
![image](https://github.com/acts-project/acts/assets/72298366/7a049bbc-51e8-42a0-9d75-49da4b12950f)